src/cast/safe/cast.rs

```diff
@@ -0,0 +1,103 @@
+/// Cast from one numeric type to another.
+///
+/// The `CastFrom<T>` trait is similar to `std::convert::From<T>`. However,
+/// while `std::convert::From<T>` performs a logical conversion, `CastFrom<T>`
+/// performs a bitwise cast from one number to another; possibly of a different
+/// size, signedness or even numeric type.
```

Comment:
I don't understand why you call this a bitwise cast -- that doesn't make any sense to me for types of different sizes.  Really you're just casting with `as`, which is also a "logical" (if sometimes lossy) conversion, especially when floating point is involved.

Overall, I don't see how `CastFrom` has any meaningful difference from `AsPrimitive`.

Reply:
Bitwise was the wrong word. Sorry. Late night. :)

`CastFrom` is a stylistic cleanup of AsPrimitive to match the other new traits if they are merged. The idea is to have identical syntax across all the conversion types.